### ISSUE TITLE
Recommend to unset associative array variable before declaration to avoid conversion

### ISSUE THREAD

**Issue by pothos** [koalaman/shellcheck#2964]
#### For new checks and feature suggestions
- [x] https://www.shellcheck.net/ (i.e. the latest commit) currently gives no useful warnings about this
- [ x I searched through https://github.com/koalaman/shellcheck/issues and didn't find anything related

#### Here's a snippet or screenshot that shows the problem:

```sh
#!/bin/bash
myarr="z"
declare -A myarr
myarr[a]=x
echo "${!myarr[@]}"
echo "${myarr[@]}"
```

This outputs
```
0 a
z x
```
because `myarr` is converted from a string `z` to having (0, `z`) as key-value pair

#### Here's what shellcheck currently says:

nothing

#### Here's what I wanted or expected to see:

`recommendation: use 'unset' before `declare` to avoid conversion of existing variable contents`
